acme/acme/client.py

```diff
@@ -245,26 +245,34 @@ def answer_challenge(self, challb, response):
 
     @classmethod
     def retry_after(cls, response, default):
-        """Compute next `poll` time based on response ``Retry-After`` header.
+        """Compute next `poll` time based on response ``Retry-After`` header,
+        per https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.37
 
         :param requests.Response response: Response from `poll`.
         :param int default: Default value (in seconds), used when
             ``Retry-After`` header is not present or invalid.
 
         :returns: Time point when next `poll` should be performed.
-        :rtype: `datetime.datetime`
+        :rtype: `datetime.datetime.datetime`
 
         """
-        retry_after = response.headers.get('Retry-After', str(default))
+        retry_after = response.headers.get('Retry-After', str(default)).strip()
         try:
             seconds = int(retry_after)
         except ValueError:
             # pylint: disable=no-member
-            decoded = werkzeug.parse_date(retry_after)  # RFC1123
-            if decoded is None:
+            when = parsedate_tz(retry_after)
+            try:
+                year = when[0] # raises TypeError if t is None
+                # py26: Handle two-digit years -- but any server that thinks
+                # "retry after 99" means "come back after 1999" is.. deprecated
+                if year >= 0 and year < 100:
+                    when = [year + 2000] + list(when[1:]) #  pragma: no cover
```

Comment:
No, cast is not required for Python 2.6. The problem here is that you're trying to concatenate list and tuple which won't work with any version of Python. You should probably use `when = (year + 2000,) + when[1:]`.